rust/src/meas/geometric/mod.tex

```diff
@@ -0,0 +1,283 @@
+        \item Type \texttt{T} must have traits \texttt{SampleTwoSidedGeometric} and \texttt{CheckNull}
+    \end{itemize}
+\end{itemize}
+
+\vicki{not sure about the domain/T stuff; it's mentioned in the implementation of the domain so I figured should also have it here} 
+
+\subsubsection*{Postconditions}
+
+\begin{itemize}
+    \item A \texttt{Measurement} is returned (i.e., if a \texttt{Measurement} cannot be returned successfully, then an error should be returned).
+\end{itemize}
+
+
+\begin{lstlisting}[language=Python,  escapechar=|]
+def make_base_geometric(scale, bounds={"lower": T_MIN, "upper": T_MAX}):
+    input_domain = AllDomain(T)
+    output_domain = AllDomain(T)
+    input_metric = L1Distance(T)
+    similarity_metric = MaxDivergence(QO)
+    
+    # check scale sign
+    if (scale < 0): |\label{line:scale}|
+        raise Exception("scale must not be negative")
+        
+    # check bounds - can discuss if python dict is best representation of this
+    if (bounds["lower"] > bounds["upper"]): |\label{line:bounds}|
+        raise Exception("lower may not be greater than upper")
+
+    def privacy_map(d_in: u_32) -> float: |\label{line:map}|
+        if (d_in < 0): |\label{line:positivedin}|
+            raise Exception("sensitivity must be non-negative")
+        if (scale == 0):
+            return QO_MAX
+        return d_in * inf_div(scale);
+
+    # these are two separate functions in original, but can discuss if we want to combine the two
+    def noise_function(arg : T, scale : QO, bounds : Optional[{T, T}]) -> Function<f64, Optional[{T, T}]>: |\label{line:noise_fn}|
+        return function(inf_cast(scale), bounds)
+        
+    def function(arg : T, scale : f64, bounds: Optional[{T, T}]) -> <D, QO>: |\label{line:fn}|
+        sample_two_sided_geometric(arg, scale, bounds)
+    
+    return Measurement(input_domain, output_domain, function, input_metric, similarity_metric, privacy_map)
+
+\end{lstlisting}
+\hanwen{Changed the input metric to L1Distance for vector
+    (was AbsoluteDistance, but I guess AbsoluteDistance could 
+    be viewed as a special case of L1Distance with dimension 1)}
+\hanwen{re. noise function, I guess it's defined separately to account for different input domain (AllDomain and VectorDomain), but everything else looks pretty much the same to me. I would vote for using just one function and call
+sample\_two\_sided\_geometric once inside, plus some brief explanation of the domain?}
+
+\vicki{may be floating point issues with \texttt{is\_sign\_negative}}
+
+\vicki{marked bounds with [SELF::MIN, SELF::MAX] because it's an optional type in the original Rust code, and if not specified, then in SampleTwoSidedGeometric SELF::MIN and SELF::MAX are substituted}
+
+\vicki{the current code does \texttt{d\_out.neg\_inf\_mul(scale)}, but since we're working with maps now, I changed that to \texttt{inf\_div}. worth asking if that's ok though} 
```

Comment:
That's what the current code does now:
https://github.com/opendp/opendp/blob/430392741ead9782304e2eda0d78384530ea1237/rust/src/meas/geometric/mod.rs#L79

It was written the way it was before because it handled zero scales without additional if-control flow. Not possible with maps, so I just added an `if`.